backend/src/v2/compiler/argocompiler/dag.go

```diff
@@ -201,7 +201,8 @@ func (c *workflowCompiler) task(name string, task *pipelinespec.PipelineTaskSpec
 				return nil, err
 			}
 			driverTaskName := name + "-driver"
-			kubernetesConfigPlaceholder, err := c.useKubernetesImpl(componentName)
+			// The following call will return an empty string for tasks without kubernetes-specific annotation.
+			kubernetesConfigPlaceholder, _ := c.useKubernetesImpl(componentName)
```

Comment:
Added a comment to explain what is expected from containers without kubernetes annotation. Also, removed the unused error.